packages/web3-eth-contract/src/types.ts

```diff
@@ -69,17 +148,30 @@ export type TransactionReceipt = ReceiptInfo;
 export interface NonPayableCallOptions {
 	nonce?: HexString;
 	chainId?: HexString;
+	/**
+	 * The address the call `transaction` should be made from. For calls the `from` property is optional however it is
+	 * highly recommended to explicitly set it or it may default to address(0) depending on your node or provider.
+	 */
 	from?: Address;
 	to?: Address;
 	data?: HexString;
+	/**
+	 * The maximum gas provided for this call “transaction” (gas limit)
+	 */
 	gas?: string;
 	maxPriorityFeePerGas?: HexString;
```

Comment:
we should add comments for type `0x2` tx fields as well

Reply:
Please suggest what you want to add.